llama_hub/tools/cogniswitch/base.py

```diff
@@ -0,0 +1,124 @@
+    ) -> dict:
+        """
+        Store data using the Cogniswitch service.
+
+        Args:
+            cs_token (str): Cogniswitch token.
+            OAI_token (str): OpenAI token.
+            url (Optional[str]): URL link.
+            file (Optional[str]): file path of your file.
+            the current files supported by the files are
+            .txt, .pdf, .docx, .doc, .html
+            document_name (Optional[str]): Name of the document you are uploading.
+            document_description (Optional[str]): Description of the document.
+
+
+
+        Returns:
+            dict: Response JSON from the Cogniswitch service.
+        """
+        if not file:
+            api_url = (
+                "https://api.cogniswitch.ai:8243/cs-api/0.0.1/cs/knowledgeSource/url"
+            )
+            headers = {
+                "apiKey": self.apiKey,
+                "openAIToken": self.OAI_token,
+                "platformToken": self.cs_token,
+            }
+            files = None
+            data = {
+                "url": url,
+                "documentName": document_name,
+                "documentDescription": document_description,
+            }
+            response = requests.post(
+                api_url, headers=headers, verify=False, data=data, files=files
+            )
+
+        if not url:
+            api_url = (
+                "https://api.cogniswitch.ai:8243/cs-api/0.0.1/cs/knowledgeSource/file"
+            )
+
+            headers = {
+                "apiKey": self.apiKey,
+                "openAIToken": self.OAI_token,
+                "platformToken": self.cs_token,
+            }
+            if file is not None:
+                files = {"file": open(file, "rb")}
+            else:
+                files = None
+            data = {
+                "url": url,
+                "documentName": document_name,
+                "documentDescription": document_description,
+            }
+            response = requests.post(
+                api_url, headers=headers, verify=False, data=data, files=files
+            )
+        if response.status_code == 200:
+            return response.json()
+        else:
+            # error_message = response.json()["message"]
+            return {
+                "message": "Bad Request",
+            }
+
+    def query_knowledge(self, query: str) -> dict:
+        """
+        Send a query to the Cogniswitch service and retrieve the response.
+
+        Args:
+            cs_token (str): Cogniswitch token.
+            OAI_token (str): OpenAI token.
+            query (str): Query to be answered.
+
+        Returns:
+            dict: Response JSON from the Cogniswitch service.
+        """
+        api_url = "https://api.cogniswitch.ai:8243/cs-api/0.0.1/cs/knowledgeRequest"
+
+        headers = {
+            "apiKey": self.apiKey,
+            "platformToken": self.cs_token,
+            "openAIToken": self.OAI_token,
+        }
```

Comment:
headers may be a function get_headers, since you're calling it 3x, having a clean code

Reply:
thanks Emanuel will update accordingly.